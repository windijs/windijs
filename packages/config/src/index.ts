export * from "./bg";
export * from "./svg";
export * from "./flex";
export * from "./grid";
export * from "./border";
export * from "./effects";
export * from "./behaviors";
export * from "./filter";
export * from "./opacity";
export * from "./sizing";
export * from "./spacing";
export * from "./animation";
export * from "./transform";
export * from "./positioning";
export * from "./interactivity";
export * from "./transitions";
export * from "./typography";
export * from "./keyframes";
export * from "./interface";
