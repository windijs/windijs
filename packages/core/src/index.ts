export * from "./base";
export * from "./api";
export * from "./handler";
export * from "./prefixer";
export * from "./builder";
