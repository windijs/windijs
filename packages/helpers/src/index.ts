// TODO: maybe add more helpers reference sass?
// https://sass-lang.com/documentation/modules

export * from "./types";
export * from "./build";
export * from "./config";
export * from "./color";
export * from "./common";
export * from "./css";
export * from "./data";
export * from "./dollar";
export * from "./escape";
export * from "./funcs";
export * from "./math";
export * from "./meta";
export * from "./namer";
export * from "./proxy";
export * from "./unit";
