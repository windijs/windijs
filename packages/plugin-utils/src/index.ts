export * from "./plugin";
export * from "./types";
export * from "./utils";
export * from "./gen";
