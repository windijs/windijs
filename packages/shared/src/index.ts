export * from "./gen";
export * from "./utils";
export * from "./types";
