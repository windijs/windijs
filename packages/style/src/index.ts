export * from "./utility";
export * from "./handler";
