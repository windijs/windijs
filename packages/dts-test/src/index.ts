import { defineConfig } from "@windijs/config";

defineConfig({
  utilities: {

  },
});
