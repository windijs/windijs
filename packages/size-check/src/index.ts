import { colorHandler, configHandler, createUtility } from "@windijs/core";

export const p = createUtility("p").use(configHandler({}, "padding")).case("t", colorHandler({}, "paddingTop")).init();
