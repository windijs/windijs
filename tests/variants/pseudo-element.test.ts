import { createUtility, backgroundColor, colors, backgroundClip, backgroundClipConfig, before, after, fileSelectorButton, firstLetter, firstLine, marker, selection } from "../../src";

const bg = createUtility()
  .use(backgroundColor(colors))
  .use(backgroundClip(backgroundClipConfig))
  .init();

const utilities = [bg.blue[500], bg.clip.content];

test("before", () => {
  expect(before(...utilities)).toMatchSnapshot();
});

test("after", () => {
  expect(after(...utilities)).toMatchSnapshot();
});

test("fileSelectorButton", () => {
  expect(fileSelectorButton(...utilities)).toMatchSnapshot();
});

test("firstLetter", () => {
  expect(firstLetter(...utilities)).toMatchSnapshot();
});

test("firstLine", () => {
  expect(firstLine(...utilities)).toMatchSnapshot();
});

test("marker", () => {
  expect(marker(...utilities)).toMatchSnapshot();
});

test("selection", () => {
  expect(selection(...utilities)).toMatchSnapshot();
});
