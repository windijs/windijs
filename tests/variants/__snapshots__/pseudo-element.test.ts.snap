// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`after 1`] = `
Object {
  "css": Object {
    "&::after": Object {
      "backgroundClip": "content-box",
      "backgroundColor": "#3b82f6",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`before 1`] = `
Object {
  "css": Object {
    "&::before": Object {
      "backgroundClip": "content-box",
      "backgroundColor": "#3b82f6",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`fileSelectorButton 1`] = `
Object {
  "css": Object {
    "&::file-selector-button": Object {
      "backgroundClip": "content-box",
      "backgroundColor": "#3b82f6",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`firstLetter 1`] = `
Object {
  "css": Object {
    "&::first-letter": Object {
      "backgroundClip": "content-box",
      "backgroundColor": "#3b82f6",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`firstLine 1`] = `
Object {
  "css": Object {
    "&::first-line": Object {
      "backgroundClip": "content-box",
      "backgroundColor": "#3b82f6",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`marker 1`] = `
Object {
  "css": Object {
    "& *::marker, &::marker": Object {
      "backgroundClip": "content-box",
      "backgroundColor": "#3b82f6",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`selection 1`] = `
Object {
  "css": Object {
    "& *::selection, &::selection": Object {
      "backgroundClip": "content-box",
      "backgroundColor": "#3b82f6",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;
